test/bun.js/fetch.test.js

```diff
@@ -1,71 +1,220 @@
 import { afterAll, beforeAll, describe, expect, it, test } from "bun:test";
 import fs, { chmodSync, unlinkSync } from "fs";
 import { mkfifo } from "mkfifo";
+import { exitCode } from "process";
 import { gc, withoutAggressiveGC } from "./gc";
 
 const exampleFixture = fs.readFileSync(
   import.meta.path.substring(0, import.meta.path.lastIndexOf("/")) + "/fetch.js.txt",
   "utf8",
 );
 
-let server ;
+let server;
 beforeAll(() => {
   server = Bun.serve({
-    async fetch(){
+    async fetch(request) {
+     
+      if (request.url.endsWith("/nodelay")) {
+        return new Response("Hello")
+      }
+      if (request.url.endsWith("/stream")) {
+        const reader = request.body.getReader();
+        const body = new ReadableStream({
+          start(controller) {
```

Comment:
this test doesn't test much because `start` is executed synchronously

instead, let's test with `pull`